app/services/fastly_config/snippets.rb

```diff
@@ -0,0 +1,62 @@
+module FastlyConfig
+  class Snippets < Base
+    FASTLY_FILES = Rails.root.join("config/fastly/snippets/*.vcl").freeze
+
+    private
+
+    def file_updated?(filename)
+      snippet_name = File.basename(filename, ".vcl").humanize
+      snippet = find_snippet(fastly, active_version, snippet_name)
+
+      return true if snippet.nil?
+
+      new_snippet_content = File.read(filename)
+      new_snippet_content != snippet.content
+    end
+
+    def update_config(new_version, filename)
+      snippet_name = File.basename(filename, ".vcl").humanize
+      snippet = find_snippet(fastly, new_version, snippet_name)
+      new_snippet_content = File.read(filename)
+
+      if snippet
+        snippet.content = new_snippet_content
+        snippet.save!
+        log_to_datadog("update", snippet, new_version)
+      else
+        snippet_options = {
+          content: new_snippet_content,
+          dynamic: 0,
+          name: snippet_name,
+          service_id: ApplicationConfig["FASTLY_SERVICE_ID"],
+          type: "init",
+          version: new_version.number
+        }
+
+        snippet = fastly.create(Fastly::Snippet, snippet_options)
+        log_to_datadog("create", snippet, new_version)
+      end
+
+      true
+    end
+
+    def find_snippet(fastly, new_version, snippet_name)
+      fastly.get_snippet(ApplicationConfig["FASTLY_SERVICE_ID"], new_version.number, snippet_name)
+    rescue Fastly::Error => e
+      error_message = JSON.parse(e.message)
+      raise e unless error_message["msg"] == "Record not found"
+
+      nil
```

Comment:
Silently swallowing all Fastly errors that are not "Record not found" makes me a bit uneasy. What other errors are possible? Should we at least log them?

Reply:
Unfortunately Fastly raises all errors as `Fastly::Error` and the devil is in the details (the message). What do you mean to log them? We re-raise all errors except that one. Do you mean to log them anyways in case that logic is ever faulty?

Reply:
1. I misread the code.
2. I'm just paranoid.